src/utils/sourcemaps/inject.rs

```diff
@@ -416,14 +419,16 @@ pub fn find_matching_paths(candidate_paths: &[String], expected_path: &str) -> V
             expected_segments.next();
         }
 
-        if candidate_segments.next().is_none() || expected_segments.next().is_none() {
-            continue;
-        }
+        // The next segments (if there are any left) must be where the paths disagree.
+        candidate_segments.next();
+        expected_segments.next();
 
         let candidate_stem = candidate_segments.collect::<Vec<_>>();
         let expected_stem = expected_segments.collect::<Vec<_>>();
 
-        if candidate_stem == expected_stem {
+        // The rest of both paths must agree and be nonempty, so at least the filenames definitely
+        // must agree.
+        if !candidate_stem.is_empty() && candidate_stem == expected_stem {
```

Comment:
```suggestion
        // The rest of both paths must agree and be nonempty, so at least the filenames definitely
        // must agree.
        if candidate_segments.peek().is_some() && itertools::equal(candidate_segments, expected_segments) {
```

As you are using `itertools` anyway, you can use [`itertools::equal`](https://docs.rs/itertools/latest/itertools/fn.equal.html) for this.

Reply:
We don’t even need that, as [`Iterator::eq`](https://doc.rust-lang.org/nightly/core/iter/trait.Iterator.html#method.eq) exists

Reply:
And by extension, the entire function doesn't need any vectors.